src/http/multipart.cr

```diff
@@ -0,0 +1,81 @@
+require "./multipart/*"
+
+module HTTP::Multipart
+  # Parses a MIME multipart message, yielding `HTTP::Headers` and an `IO` for
+  # each body part.
+  #
+  # Please note that the IO object yielded to the block is only valid while the
+  # block is executing. The IO is closed as soon as the supplied block returns.
+  #
+  # ```
+  # multipart = "--aA40\r\nContent-Type: text/plain\r\n\r\nbody\r\n--aA40--"
+  # HTTP::Multipart.parse(IO::Memory.new(multipart), "aA40") do |headers, io|
+  #   headers["Content-Type"] # => "text/plain"
+  #   io.gets_to_end          # => "body"
+  # end
+  # ```
+  #
+  # See: `Multipart::Parser`
+  def self.parse(io, boundary)
+    parser = Parser.new(io, boundary)
+    while parser.has_next?
+      parser.next { |headers, io| yield headers, io }
```

Comment:
This is plain multipart, not multipart/form-data. Field names and file metadata are specific to the `multipart/form-data` mime-type.